### ISSUE TITLE
MD009 for list continuation

### ISSUE THREAD

**Issue by CircleCode** [DavidAnson/markdownlint#55]
when you make a multi paragraphs list, you have to continue indentation on empty lines. In this case, rule MD009 report an error.
To solve this issue, MD009 should not trigger errors for a line with only spaces if
-   previous lines starts by at least the same amount of lines and following line starts by exactly the same amount of spaces
-   or previous lines starts by exactly the same amount of lines and following line starts by at least the same amount of spaces

Example:
```
-   first paragraph of the item
    
    second paragraph of the same item
```
(note: although this is not visible, line 2 is exactly 4 paragraphs)

**Comment by DavidAnson** [comment]
Please refer to section 5.2 of the specification, specifically example 217: http://spec.commonmark.org/0.27/#list-items

Blank lines are not required to contain a matching number of spaces. You can verify this into practice using the  underlying parser here: https://markdown-it.github.io/

**Comment by DavidAnson** [comment]
If we agree on that, I suggest it is preferable to avoid trailing whitespace for such blank lines and that the current behavior of this rule is reasonable.

**Comment by CircleCode** [comment]
I agree with the fact that the current behavior is reasonable regarding commonmark.
If your goal is to strictly adhere to commonmark, this issue can be closed.

However, I think a lot of still living implementations require this indentation continuation.
If you don not disagree with the idea of supporting those implementations, but are reluctant to make a lot of work for them (which makes sense), maybe simply adding an option to ignore lines with only spaces for rule MD009 would be acceptable?

In any case, I'll respect your choice.

TBH, I'd love to switch to commonmark, but while there is no extension mechanism, this is too restricting for my needs.

**Comment by DavidAnson** [comment]
Your option to ignore for blank lines seems like a good compromise, but it may not be all that difficult to implement this according to your original proposal. Thank you for the suggestion and feedback!
